### PR TITLE
Add helper for routing and asserting CE replies in E2E tests

### LINKED ISSUE #536: Create a test helper for intercepting and asserting event responses
I suggest that we implement a helper inside the E2E test framework for automatically deploying a Knative Eventing topology that would allow us to intercept and assert event responses from any Addressable.

The helper would deploy, and auto-wire, the 3 components below:
- a `Channel`, which acts as the entry point for routing CloudEvents to the Addressable
- an event-display `Deployment` (+k8s `Service` in front of it), which receives responses
- a `Subscription`, which performs the actual routing of incoming events to the Addressable, and routing of responses to event-display

The helper returns the following values:
- the address of the `Channel` (or a reference we can use to read the address), to be able to send events to it
- a reference to the event-display `Deployment`, so we can read the events it received

E2E reply analysis drawio

### PR DESCRIPTION
**Pull request by antoineco**
Closes #536

**tl;dr;** Allows us to fairly easily test event **replies** in any component by routing these to event-display using a Knative Channel + Subscription combo.

Before:

```
                 ┌──────────────────┐
test event ─────►│                  │
                 │ XMLToJSONTransf. │
  (?) <----------│                  │
         replies └──────────────────┘
```

After:

```
                 ┌─────────┐ ┌──────────────┐         ┌──────────────────┐
                 │         │ │              ├─────────►                  │
test event ─────►│ Channel │o│ Subscription │         │ XMLToJSONTransf. │
                 │         │ │             ┌┤---------│                  │
                 └─────────┘ └─────────────┴┘ replies └──────────────────┘
                                           |
                                           |
                                           V
                                      ┌───────────────┐
                                      │ event-display │
                                      └───────────────┘
```

---

@JeffNeff as part of this PR, I wrote the unfinished part of the `XMLToJSONTransformation` E2E test (remember #484?), which repeats the existing test but without `K_SINK`.
It works smoothly and the CE replies are delivered to event-display as expected:

```jsonc
☁️  cloudevents.Event
Context Attributes,
  specversion: 1.0
  type: e2e.test
  source: e2e.triggermesh
  id: 0000
  time: 2022-04-06T08:54:40.830166Z
  datacontenttype: application/json
Extensions,
  iotriggermeshe2e: e2e-xmltojsontransformation-4723
Data,
  {
    "note": {
      "body": "Dont forget me this weekend",
      "to": "Tove",
      "from": "Jani",
      "heading": "Reminder"
    }
  }

```

### CODE REVIEW

**Comment by antoineco** [comment]
Closing and reopening because CircleCI missed the PR due to an outage.
